oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/SharedStorageApiLiveTest.java

```diff
@@ -0,0 +1,152 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.oneandone.rest.features;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.jclouds.oneandone.rest.domain.Server;
+import org.apache.jclouds.oneandone.rest.domain.SharedStorage;
+import org.apache.jclouds.oneandone.rest.domain.SharedStorageAccess;
+import org.apache.jclouds.oneandone.rest.domain.Types;
+import org.apache.jclouds.oneandone.rest.domain.options.GenericQueryOptions;
+import org.apache.jclouds.oneandone.rest.internal.BaseOneAndOneLiveTest;
+import org.testng.Assert;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "SharedStorageApiLiveTest")
+public class SharedStorageApiLiveTest extends BaseOneAndOneLiveTest {
+
+   private SharedStorage currentSharedStorage;
+   private Server currentServer;
+   private List<SharedStorage> sharedStorages;
+
+   private SharedStorageApi sharedStorageApi() {
+
+      return api.sharedStorageApi();
+   }
+
+   @BeforeClass
+   public void setupTest() {
+      currentServer = createServer("sharestorage jclouds server");
+      assertNodeAvailable(currentServer);
+      currentSharedStorage = sharedStorageApi().create(SharedStorage.CreateSharedStorage.builder()
+              .name("jcloudsStorage")
+              .description("desc")
+              .size(50)
+              .build());
+      assertNotNull(currentSharedStorage);
+   }
+
+   @Test
+   public void testList() {
+      sharedStorages = sharedStorageApi().list();
+
+      assertNotNull(sharedStorages);
+      Assert.assertTrue(sharedStorages.size() > 0);
+   }
+
+   public void testListWithOption() {
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, "jcloudsStorage", null);
+      List<SharedStorage> imageWithQuery = sharedStorageApi().list(options);
+
+      assertNotNull(imageWithQuery);
+      Assert.assertTrue(imageWithQuery.size() > 0);
+   }
+
+   public void testGet() {
+      SharedStorage result = sharedStorageApi().get(currentSharedStorage.id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentSharedStorage.id());
+   }
+
+   @Test(dependsOnMethods = "testGet")
+   public void testUpdate() throws InterruptedException {
+      String updatedName = "Updatedjava";
+
+      SharedStorage updateResult = sharedStorageApi().update(currentSharedStorage.id(), SharedStorage.UpdateSharedStorage.create(updatedName, "desc", null));
+
+      assertNotNull(updateResult);
+      assertEquals(updateResult.name(), updatedName);
+
+   }
+
+   @Test(dependsOnMethods = "testUpdate")
```

Comment:
Looks like it could run as the first test after the setup, as it just needs the shared storage. Is there a real need for this dependency?